File: trunk/TeRKPeerCommon/code/c#/src/TeRK/components/services/AnalogInputsService.cs
using TeRK.services;

namespace TeRK.components.services
   {
   public abstract class AnalogInputsService : Service
      {
      public const string TYPE_ID = "::TeRK::AnalogInController";

      public string getTypeId()
         {
         return TYPE_ID;
         }

      /** Returns the state of the analog inputs. */
      public abstract AnalogInState getAnalogInState();
      }
   }

File: trunk/TeRKPeerCommon/code/java/analogin/src/edu/cmu/ri/createlab/TeRK/analogin/AnalogInputsService.java
package edu.cmu.ri.createlab.TeRK.analogin;

import edu.cmu.ri.mrpl.TeRK.services.DeviceController;
import edu.cmu.ri.mrpl.TeRK.services.Service;

/**
 * @author Chris Bartley (bartley@cmu.edu)
 */
public interface AnalogInputsService extends Service, DeviceController
   {
   String TYPE_ID = "::TeRK::AnalogInController";

   /**
    * Returns the value of all analog inputs or <code>null</code> if the values could not be retrieved.
    */
   short[] getAnalogInputValues();

   /**
    * Returns the value of the given port id.
    *
    * @throws NullPointerException if values could not be retrieved
    * @throws IndexOutOfBoundsException if the <code>analogInputPortId</code> specifies an invalid port
    */
   short getAnalogInputValue(final int analogInputPortId);
   }
